File: details/src/main/java/com/nikita/popularmovies/details/ReviewsAdapter.java
package com.nikita.popularmovies.details;

import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.nikita.popularmovies.Views;
import com.nikita.popularmovies.common.models.Review;

import java.util.Collections;
import java.util.List;

public final class ReviewsAdapter extends PagerAdapter {
  private List<Review> reviews = Collections.emptyList();

  public void changeData(List<Review> data) {
    reviews = data;
    notifyDataSetChanged();
  }

  @Override
  public Object instantiateItem(ViewGroup container, int position) {
    View view = LayoutInflater.from(container.getContext()).inflate(R.layout.movie_details_review_item, container, false);

    TextView author = Views.findView(view, R.id.author);
    TextView content = Views.findView(view, R.id.content);
    Review review = reviews.get(position);
    author.setText(review.author);
    content.setText(review.content);

    container.addView(view);
    return view;
  }

  @Override
  public void destroyItem(ViewGroup container, int position, Object object) {
    container.removeView((View) object);
  }

  @Override
  public int getCount() {
    return reviews.size();
  }

  @Override
  public boolean isViewFromObject(View view, Object object) {
    return view == object;
  }
}

File: details/src/main/kotlin/com/nikita/popularmovies/details/ReviewsAdapter.kt
package com.nikita.popularmovies.details

import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.nikita.popularmovies.common.findView
import com.nikita.popularmovies.common.models.Review

class ReviewsAdapter(private var reviews: List<Review> = emptyList()) : PagerAdapter() {
  fun changeData(data: List<Review>) {
    reviews = data
    notifyDataSetChanged()
  }

  override fun instantiateItem(container: ViewGroup, position: Int): Any {
    val view = LayoutInflater.from(container.context).inflate(R.layout.movie_details_review_item, container, false)

    val author: TextView = view.findView(R.id.author)
    val content: TextView = view.findView(R.id.content)
    val review = reviews[position]
    author.text = review.author
    content.text = review.content

    container.addView(view)
    return view
  }

  override fun destroyItem(container: ViewGroup, position: Int, obj: Any) {
    container.removeView(obj as View)
  }

  override fun isViewFromObject(view: View, obj: Any) = view == obj

  override fun getCount() = reviews.size
}